contrib/ansible/library/skydive_edge_list.py

```diff
@@ -0,0 +1,216 @@
+#
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: skydive_edge
+
+short_description: Module to add edges to Skydive topology
+
+version_added: "1.0"
+
+description:
+    - "This module handles edge adding to the Skydive topology"
+
+options:
+    analyzer:
+        description:
+            - analyzer address
+        default: localhost:8082
+        required: true
+    ssl:
+        description:
+            - SSL enabled
+    insecure:
+        description:
+            - ignore SSL certification verification
+    username:
+        description:
+            - username authentication parameter
+    password:
+        description:
+            - password authentication parameter
+    edges:
+        description:
+            - list of the edges. Each edge must contain node1,
+              node2 and relation_type fields
+        required: true
+    host:
+        description:
+            - host of the node
+
+notes:
+  - Using list reduces the communication overhead for creating edges.
+  - Node definitions must be UUIDs. Gremlin queries require a communication
+    roundtrip.
+
+author:
+    - Sylvain Afchain (@safchain)
```

Comment:
You should maybe change this line :)

Reply:
I will add a line as secondary author, my changes are minor compared to your code. This is one of the thing I do not like much about this code: there is a lot of boiler plate shared between modules. But it seems to be hard to share code between modules in ansible without putting constraints on the user (https://groups.google.com/forum/#!topic/ansible-devel/Tv_9GXnDJRI).

Reply:
And it was a such excellent code :) Yes not easy to share code between modules it seems. Maybe we could add helpers into the python lib directly but that is something we can address later. Thanks for the PR.